### PR TITLE
Doc for installation

### PR DESCRIPTION
**Pull request by liuzhe-lz**
### Checklist ###
  - [ ] test case
  - [ ] doc

### CODE REVIEW

File: docs/source/build_from_source.rst [hunk 1]
```diff
@@ -0,0 +1,113 @@
+Build from Source
```
Comment (ultmaster) [inline docs/source/build_from_source.rst:1]: Suggest putting it into notes/

File: docs/source/build_from_source.rst [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+.. code-block::
+
+    python -m pip install --upgrade setuptools pip wheel
+
+.. _source code: https://github.com/microsoft/nni
+
+Development Build
+-----------------
+
+If you want to build NNI for your own use, we recommend using `development mode`_.
```
Comment (ultmaster) [inline docs/source/build_from_source.rst:17]: There is also a step-by-step guide in contribution guide. We might have a cross-reference.

File: setup_ts.py [hunk 3]
```diff
@@ -15,6 +15,7 @@
 import json
 import os
 from pathlib import Path
+import platform
```
Comment (QuanluZhang) [inline setup_ts.py:18]: why the changes to this file are merged to the branch doc-refactor?

File: docs/source/build_from_source.rst (hunk 1 above)
Comment (cruiseliu) [reply docs/source/build_from_source.rst:1]: I can't find the directory?

File: setup_ts.py (hunk 3 above)
Comment (cruiseliu) [reply setup_ts.py:18]: The changes are meant to make the doc clear.
I don't want to discuss in doc that "you need to add --all because blahblah", or "to install on M1 first change line 123 to blahblah".

File: docs/source/build_from_source.rst (hunk 2 above)
Comment (cruiseliu) [reply docs/source/build_from_source.rst:17]: I have just thought about it but can't find a proper place to insert the line.
It reads like "I just want to install NNI, why do you tell me to contribute?"
I recommend develop install here merely because it's much simpler than bdist_wheel. And develop requirements are not needed if the user does not plan to make PR.
I have put this doc right above contribution guide, so the bottom bar will say "Next: How to Contribute". Maybe that's enough?

Comment (ultmaster) [reply docs/source/build_from_source.rst:17]: That's on master branch. :(
